File: VB/11_Images/SetImageFrameFormat.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Spire.Presentation
Imports Spire.Presentation.Drawing

Namespace SetImageFrameFormat
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs)
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load an image
			Dim imageFile As String = "..\..\..\..\..\..\Data\iceblueLogo.png"
			Dim image As Image = Image.FromFile(imageFile)

			'Add the image in document
			Dim imageData As IImageData = presentation.Images.Append(image)

			'////////////////Use the following code for netstandard dlls/////////////////////////
'            
'            FileStream fileStream = new FileStream(imageFile, FileMode.Open, FileAccess.Read, FileShare.Read);
'            byte[] bytes = new byte[fileStream.Length];
'            fileStream.Read(bytes, 0, bytes.Length);
'            fileStream.Close();
'            Stream stream = new MemoryStream(bytes);
'            IImageData imageData = presentation.Images.Append(stream);
'            stream.Close();
'            fileStream.Close();
'            

			Dim rect As New RectangleF(100,100,imageData.Width\2,imageData.Height\2)
			Dim pptImage As IEmbedImage = presentation.Slides(0).Shapes.AppendEmbedImage(ShapeType.Rectangle, imageData, rect)

			'Set the formatting of the image frame
			pptImage.Line.FillFormat.FillType = FillFormatType.Solid
			pptImage.Line.FillFormat.SolidFillColor.Color = Color.LightBlue
			pptImage.Line.Width = 5
			pptImage.Rotation = -45

			'Save the document
			Dim result As String = "SetImageFrameFormat_result.pptx"
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the file
			OutputViewer(result)
		End Sub
		Private Sub OutputViewer(ByVal filename As String)
			Try
				System.Diagnostics.Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/11_Images/SetImageFrameFormat.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Drawing;

namespace SetImageFrameFormat
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load an image
            string imageFile = @"..\..\..\..\..\..\Data\iceblueLogo.png";
            Image image = Image.FromFile(imageFile);

            //Add the image in document
            IImageData imageData = presentation.Images.Append(image);
            RectangleF rect = new RectangleF(100,100,imageData.Width/2,imageData.Height/2);
            IEmbedImage pptImage = presentation.Slides[0].Shapes.AppendEmbedImage(ShapeType.Rectangle, imageData, rect);

            //Set the formatting of the image frame
            pptImage.Line.FillFormat.FillType = FillFormatType.Solid;
            pptImage.Line.FillFormat.SolidFillColor.Color = Color.LightBlue;
            pptImage.Line.Width = 5;
            pptImage.Rotation = -45;

            //Save the document
            string result = "SetImageFrameFormat_result.pptx";
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the file
            OutputViewer(result);
        }
        private void OutputViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}